src/query/query_opfunc.c

```diff
@@ -8699,7 +8699,7 @@ qdata_get_class_of_function (THREAD_ENTRY * thread_p, FUNCTION_TYPE * function_p
     }
 
   instance_oid_p = DB_PULL_OID (val_p);
-  if (heap_get_class_oid (thread_p, &class_oid, instance_oid_p) != S_SUCCESS)
+  if (heap_get_class_oid (thread_p, instance_oid_p, &class_oid) != S_SUCCESS)
```

Comment:
I am not sure it is correct to call heap_get_class_oid here. This is class_of function. However, if object is null (deleted), my expectation is that class_of will fail or will return NULL.

I think prior to your change, heap_get_class_oid function would have checked object. If the object is deleted, or invisible, the result would not be S_SUCCESS. The result is ER_FAILED, the error is propagated and the query fails.

With your change, reading class oid from page does not fail and the query will continue successfully.